src/main/java/fr/xephi/authme/command/executable/authme/RegisterAdminCommand.java

```diff
@@ -51,38 +51,31 @@ public void executeCommand(final CommandSender sender, List<String> arguments) {
             return;
         }
 
-        bukkitService.runTaskOptionallyAsync(new Runnable() {
-
-            @Override
-            public void run() {
-                if (dataSource.isAuthAvailable(playerNameLowerCase)) {
-                    commonService.send(sender, MessageKey.NAME_ALREADY_REGISTERED);
-                    return;
-                }
-                HashedPassword hashedPassword = passwordSecurity.computeHash(playerPass, playerNameLowerCase);
-                PlayerAuth auth = PlayerAuth.builder()
-                    .name(playerNameLowerCase)
-                    .realName(playerName)
-                    .password(hashedPassword)
-                    .build();
+        bukkitService.runTaskOptionallyAsync(() -> {
+            if (dataSource.isAuthAvailable(playerNameLowerCase)) {
+                commonService.send(sender, MessageKey.NAME_ALREADY_REGISTERED);
+                return;
+            }
+            HashedPassword hashedPassword = passwordSecurity.computeHash(playerPass, playerNameLowerCase);
+            PlayerAuth auth = PlayerAuth.builder()
+                .name(playerNameLowerCase)
+                .realName(playerName)
+                .password(hashedPassword)
+                .build();
 
-                if (!dataSource.saveAuth(auth)) {
-                    commonService.send(sender, MessageKey.ERROR);
-                    return;
-                }
-                dataSource.setUnlogged(playerNameLowerCase);
+            if (!dataSource.saveAuth(auth)) {
+                commonService.send(sender, MessageKey.ERROR);
+                return;
+            }
+            dataSource.setUnlogged(playerNameLowerCase);
+            dataSource.revokeSession(playerNameLowerCase);
```

Comment:
are the above two calls necessary as a player gets saved? I'd expect those are the defaults.

Reply:
Right, good point